src/bin_node/node_storage_command.ml

```diff
@@ -0,0 +1,149 @@
+(*****************************************************************************)
+(*                                                                           *)
+(* Open Source License                                                       *)
+(* Copyright (c) 2020 Tarides <contact@tarides.com>                          *)
+(*                                                                           *)
+(* Permission is hereby granted, free of charge, to any person obtaining a   *)
+(* copy of this software and associated documentation files (the "Software"),*)
+(* to deal in the Software without restriction, including without limitation *)
+(* the rights to use, copy, modify, merge, publish, distribute, sublicense,  *)
+(* and/or sell copies of the Software, and to permit persons to whom the     *)
+(* Software is furnished to do so, subject to the following conditions:      *)
+(*                                                                           *)
+(* The above copyright notice and this permission notice shall be included   *)
+(* in all copies or substantial portions of the Software.                    *)
+(*                                                                           *)
+(* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR*)
+(* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,  *)
+(* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL   *)
+(* THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER*)
+(* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING   *)
+(* FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER       *)
+(* DEALINGS IN THE SOFTWARE.                                                 *)
+(*                                                                           *)
+(*****************************************************************************)
+
+let term_name = "storage"
+
+module Term = struct
+  open Cmdliner
+
+  (* [Cmdliner] terms are not nestable, so we implement an ad-hoc mechanism for
+     delegating to one of several "subcommand"s by parsing a single positional
+     argument and then calling [Term.eval] again with the remaining
+     arguments. *)
+
+  type subcommand = {
+    name : string;
+    description : string;
+    term : (unit -> unit) Term.t;
+  }
+
+  let terms =
+    let open Context.Checks in
+    [ {
+        name = "stat-pack";
+        description = "Print high-level statistics about the pack file.";
+        term = Pack.Stat.term;
+      };
+      {
+        name = "stat-index";
+        description = "Print high-level statistics about the index store.";
+        term = Index.Stat.term;
+      };
+      {
+        name = "integrity-check";
```

Comment:
I would rename it in `integrity-check-index`, as we have an integrity check in pack too, that I think eventually we will expose to the cli as well. 
Also, I'm wondering if it's not useful to have a comment somewhere to specify that commands ending with "-index" need the path to the index (and maybe include that its usually `/data/context/` or `/data/context/upper0/`)? I see that `./tezos-node --help` has an example section, it could go there.

Reply:
Re. `integrity-check-index`: agreed, thanks :+1:

Re. adding examples: I think I'd rather leave the documentation fairly minimal since these subcommands are still quite unstable. For instance, it probably _is_ best if they share the same `--data-dir` semantics as the other commands, but that's not worthwhile at this stage. In fact, I think it's probably best to add some more warnings about this being an unstable API; I'll do that.